services.ts

```diff
@@ -18,7 +18,9 @@ export const checkToken = async (req: Request, res: Response) => {
 };
 
 export const createLink = async (req: Request, res: Response) => {
-    const { url, desired, mandate, expires } = req.body;
+    const { url, mandate, expires } = req.body;
+    // force lowercase on short name
+    const desired = req.body.desired.toLowerCase();
```

Comment:
This will still error, since `undefined` is a valid value for `desired` by design (in the case where we don't ask for a specific url (i.e. just 4 random chars)).

Would suggest either 

```suggestion
    const desired = req.body.desired?.toLowerCase();
```
or doing the conversion inside the if-statement at row 55. (looks nicer imo to get `desired` with destructuring)

Reply:
Oh right... My fault for not testning enough (but in my defense it is *literally impossible* to get this running locally)